arangod/Aql/IResearchViewNode.cpp

```diff
@@ -858,15 +858,13 @@ void IResearchViewNode::planNodeRegisters(
     std::vector<aql::RegisterId>& nrRegsHere, std::vector<aql::RegisterId>& nrRegs,
     std::unordered_map<aql::VariableId, aql::VarInfo>& varInfo,
     unsigned int& totalNrRegs, unsigned int depth) const {
-  nrRegsHere.emplace_back(1);
+  nrRegsHere.emplace_back(isLateMaterialized()? 0 : 1);
+
   // create a copy of the last value here
   // this is requried because back returns a reference and emplace/push_back
   // may invalidate all references
-  aql::RegisterId const registerId = 1 + nrRegs.back();
-  nrRegs.emplace_back(registerId);
-
-  varInfo.emplace(_outVariable->id, aql::VarInfo(depth, totalNrRegs++));
-
+  aql::RegisterCount const prevRegistersCount = nrRegs.back();
+  nrRegs.emplace_back(prevRegistersCount);
```

Comment:
Is this correct (omitting the +1), or does it essentially have to be `prevRegistersCount + nrRegsHere.back()`?